components/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/impl/IdentityStoreImpl.java

```diff
@@ -1731,19 +1731,20 @@ private AuthenticationContext getAuthenticationContext(AuthenticationContext aut
                                                            Domain domain)
             throws IdentityStoreException {
 
-        DomainUser domainUser = (DomainUser) authenticationContext.getParameter(IdentityMgtConstants.DOMAIN_USER);
+        String domainUserId = (String) authenticationContext.getParameter(IdentityMgtConstants.DOMAIN_USER_ID);
+        String state = (String) authenticationContext.getParameter(IdentityMgtConstants.STATE);
```

Comment:
Shall we move this inside the `if` block ? In an authentication failure, this param read is unnecessary.